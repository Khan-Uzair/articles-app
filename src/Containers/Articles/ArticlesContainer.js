import React, { Component } from 'react';
import { connect } from "react-redux";
import { fetch_articles, delete_article } from "../../Actions/ApiActions";
import { selected_article } from '../../Actions/StoreActions';
import ArticlesComponent from "./ArticlesComponent";

import toastr from 'toastr';
import LoaderComponent from '../../Components/Loader/LoaderComponent';

class ArticlesContainer extends Component {
    state = {
        headers: [
		    "Title",
		    "Description",
		    "Author",
		    "Actions"
		  ]
    }
    componentDidMount = () => {
        this.loadArticles();
    }

    loadArticles = async () => {
        await this.props.dispatch(fetch_articles());
    }

    deleteArticleHandler = async (row) => {
        const response= await this.props.dispatch(delete_article(row.id));
        if(!response) {
            toastr.error('Delete error');
        } else {
            toastr.success('Delete success');
            this.loadArticles();
        }
    }

    editArticleHandler = (row) => {
        this.props.dispatch(selected_article(row.id));
        this.props.history.push("/edit-article");
    }

    articleDetailshandler = (row) => {
        this.props.dispatch(selected_article(row.id));
        this.props.history.push("/article-details");
    }

    onActionHandler = (event, action) => {
        switch(action.type) {
            case 'edit': 
                this.editArticleHandler(action.data);
                break;
            case 'delete':
                this.deleteArticleHandler(action.data);
                break; 
            case 'details':
                this.articleDetailshandler(action.data);
                break;
            default: ;
        }
    }

    render() {
        const { data } = this.props;
        return(
            <>
                {data.isLoading ? <LoaderComponent /> : null}
                <ArticlesComponent 
                    headers={this.state.headers} 
                    articles={this.props.data}
                    onAction={this.onActionHandler}>
                </ArticlesComponent>
            </>
        );
    }
}

const mapStateToProps = state => {
    return {
      data: state
    };
};

export default connect(mapStateToProps)(ArticlesContainer);
