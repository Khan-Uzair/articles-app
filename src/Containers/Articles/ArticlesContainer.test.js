import React from 'react';
import Adapter from 'enzyme-adapter-react-16'
import { configure,shallow } from 'enzyme'
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import ArticlesContainer from './ArticlesContainer';

const mockStore = configureMockStore();
const store = mockStore({});
configure({ adapter: new Adapter() })

describe('<ArticleContainer />',()=>{
    it('should render one ArticlesComponent',()=>{
        const articleContainerElement = shallow((
            <Provider store={store}>
                <ArticlesContainer />
            </Provider>));
            expect(articleContainerElement).toMatchSnapshot();
    });
});