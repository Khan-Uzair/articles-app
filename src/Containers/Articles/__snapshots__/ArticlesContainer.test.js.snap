// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleContainer /> should render one ArticlesComponent 1`] = `ShallowWrapper {}`;
